### PR TITLE
Update z-stream to trigger jobs for rosa versions

### PR DESCRIPTION
**Pull request by chetna14manku**
Link to the ticket: https://issues.redhat.com/browse/MSINTEROP-1526

- Update zstream code to skip version, if not found instead of giving error.
- Update zstream code to trigger jobs for rosa-versions
- For provided rosa versions, add functionality to check if that version is enabled in OCM or not 
- Requires https://github.com/RedHatQE/rosa-python-client/pull/270 for rosa-versions function

### CODE REVIEW

File: ci_jobs_trigger/libs/openshift_ci/zstream_trigger/zstream_trigger.py [hunk 1]
```diff
@@ -55,6 +58,65 @@ def already_processed_version(
...
+def get_gitlab_project_files(config: Dict, ocm_env: str):
+    api = get_gitlab_api(url=config["gitlab_url"], token=config["gitlab_token"])
+    project = api.projects.get(config["gitlab_project"])
+    project_file_content = project.files.get(
+        file_path=f"config/{'prod' if ocm_env == 'production' else ocm_env}.yaml", ref="master"
+    )
+    return yaml.safe_load(project_file_content.decode().decode("utf-8"))
+
+
+def is_rosa_version_enabed(config: Dict, version: str, channel: str, ocm_env: str, logger: logging.Logger) -> bool:
```
Comment (myakove) [inline ci_jobs_trigger/libs/openshift_ci/zstream_trigger/zstream_trigger.py:70]: `enabed` typo?

File: ci_jobs_trigger/libs/openshift_ci/zstream_trigger/zstream_trigger.py [hunk 2]
```diff
@@ -141,37 +203,65 @@ def process_and_trigger_jobs(logger: logging.Logger, version: str | None = None)
                 trigger_res[_version] = "No jobs found"
                 continue
 
+            _rosa_env: str = ""
+            if "___" in _version:
```
Comment (myakove) [inline ci_jobs_trigger/libs/openshift_ci/zstream_trigger/zstream_trigger.py:207]: Please add a comment that explain why you check for `___`

File: ci_jobs_trigger/libs/openshift_ci/zstream_trigger/zstream_trigger.py [hunk 3]
```diff
@@ -141,37 +203,65 @@ def process_and_trigger_jobs(logger: logging.Logger, version: str | None = None)
...
+            _rosa_channel = "candidate" if _rosa_env and _version_channel in ["rc", "ec"] else _version_channel
+
+            if _rosa_env and config.get("gitlab_project"):
+                if not is_rosa_version_enabed(
+                    config=config, version=_wanted_version, channel=_rosa_channel, ocm_env=_rosa_env, logger=logger
+                ):
+                    logger.info(
+                        f"{LOG_PREFIX} Version {_wanted_version}:{_version_channel} not enabled for ROSA {_rosa_env}, skipping"
+                    )
+                    trigger_res[_base_version] = "Not enabled for ROsA"
```
Comment (myakove) [inline ci_jobs_trigger/libs/openshift_ci/zstream_trigger/zstream_trigger.py:226]: `ROsA` typo

**Review by myakove — CHANGES_REQUESTED** [review]

File: ci_jobs_trigger/libs/openshift_ci/zstream_trigger/zstream_trigger.py (hunk 1 above)
Comment (chetna14manku) [reply ci_jobs_trigger/libs/openshift_ci/zstream_trigger/zstream_trigger.py]: Updated

**Review by myakove — DISMISSED** [review]

File: ci_jobs_trigger/libs/openshift_ci/zstream_trigger/zstream_trigger.py [hunk 4]
```diff
@@ -89,8 +89,7 @@ def is_rosa_version_enabled(config: Dict, version: str, channel: str, ocm_env: s
 
 
 def filter_rosa_versions_by_channel(all_rosa_versions: Dict, rosa_channel: str, version_channel: str) -> Dict:
-    filtered_rosa_dict: Dict = {}
-    filtered_rosa_dict[version_channel] = {}
+    filtered_rosa_dict: Dict = {version_channel: {}}
```
Comment (myakove) [inline ci_jobs_trigger/libs/openshift_ci/zstream_trigger/zstream_trigger.py:92]: Please state the Dict types
Dict[str, .....]

File: ci_jobs_trigger/libs/openshift_ci/zstream_trigger/zstream_trigger.py [hunk 5]
```diff
@@ -239,7 +238,15 @@ def process_and_trigger_jobs(logger: logging.Logger, version: str | None = None)
                 if _rosa_env
                 else get_accepted_cluster_versions()
             )
-            _latest_version = _all_versions.get(_version_channel)[_wanted_version][0]
+
+            if not (wanted_version_list := _all_versions.get(_version_channel).get(_wanted_version)):
```
Comment (myakove) [inline ci_jobs_trigger/libs/openshift_ci/zstream_trigger/zstream_trigger.py:242]: This `_all_versions.get(_version_channel)` can fail ,you need to set default 
`_all_versions.get(_version_channel, {})`

**Review by myakove — CHANGES_REQUESTED** [review]

File: ci_jobs_trigger/libs/openshift_ci/zstream_trigger/zstream_trigger.py (hunk 5 above)
Comment (chetna14manku) [reply ci_jobs_trigger/libs/openshift_ci/zstream_trigger/zstream_trigger.py]: Updated

**Review by myakove — APPROVED** [review]

**Pull request merged**